src/api/store/misc.py

```diff
@@ -271,3 +272,38 @@ def backfill_tag_data(self, context: Context, args):
     except Exception as e:
       capture_message(str(e), level="error")
       return None, CustomMassenergizeError(e)
+  
+  
+  def create_carbon_equivalency(self, args):
+    try:
+      new_carbon_equivalency = CarbonEquivalency.objects.create(**args)
+      new_carbon_equivalency.save()
+      return None , None
+
+    except Exception as e:
+      capture_message(str(e), level="error")
+      return None, CustomMassenergizeError(e)
+
+
+  def update_carbon_equivalency(self, tag_id, args):
+    carbon_equivalency = CarbonEquivalency.objects.filter(id=tag_id)
```

Comment:
Probably needs a change relating to the icon ForeignKey, not a problem I can do that.
Or you could copy what is done is other update_ methods where there are media files.

Reply:
I'll see to this today. Do you know where I might find a good example?

Reply:
`carbon_equivalencies = CarbonEquivalency.objects.filter(id=tag_id)

      if not carbon_equivalencies:
          return None, InvalidResourceError()

      carbon_equivalencies.update(**args)
      carbon_equivalency = carbon_equivalencies.first()

      value = args.get('value', None)
      carbon_equivalency.value = value

      explanation = args.get('explanation', None)
      carbon_equivalency.explanation = explanation

      reference = args.get('reference', None)
      carbon_equivalency.reference = reference

      icon = args.get("icon", None)
      carbon_equivalency.icon = icon
      
      carbon_equivalency.save()
      
      return carbon_equivalency, None`

Reply:
`
carbon_equivalencies = CarbonEquivalency.objects.filter(id=tag_id)

      if not carbon_equivalencies:
          return None, InvalidResourceError()

      carbon_equivalencies.update(**args)
      carbon_equivalency = carbon_equivalencies.first()

      value = args.get('value', None)
      carbon_equivalency.value = value

      explanation = args.get('explanation', None)
      carbon_equivalency.explanation = explanation

      reference = args.get('reference', None)
      carbon_equivalency.reference = reference

      icon = args.get("icon", None)
      carbon_equivalency.icon = icon
      
      carbon_equivalency.save()
      
      return carbon_equivalency, None
`

Reply:
Here's what I have os far. I'm basing this off the store/userprofile.py.
`carbon_equivalencies = CarbonEquivalency.objects.filter(id=tag_id)

      if not carbon_equivalencies:
          return None, InvalidResourceError()

      carbon_equivalencies.update(**args)
      carbon_equivalency = carbon_equivalencies.first()

      value = args.get('value', None)
      carbon_equivalency.value = value

      explanation = args.get('explanation', None)
      carbon_equivalency.explanation = explanation

      reference = args.get('reference', None)
      carbon_equivalency.reference = reference

      icon = args.get("icon", None)
      carbon_equivalency.icon = icon
      
      carbon_equivalency.save()
      
      return carbon_equivalency, None`

Reply:
How is this? I think this is right. I'm using create_community as an example.
`def update_carbon_equivalency(self, tag_id, args):
    try:
      carbon_equivalencies = CarbonEquivalency.objects.filter(id=tag_id)

      if not carbon_equivalencies:
          return None, InvalidResourceError()

      carbon_equivalencies.update(**args)
      carbon_equivalency = carbon_equivalencies.first()

      value = args.get('value', None)
      if value:
        carbon_equivalency.value = value

      explanation = args.get('explanation', None)
      if explanation:
        carbon_equivalency.explanation = explanation

      reference = args.get('reference', None)
      if reference:
        carbon_equivalency.reference = reference

      icon = args.get("icon", None)
      if icon:
        cFav = Media(file=icon, name=f"{args.get('name', '')} CarbonEquivalencyFavicon")
        cFav.save()
        carbon_equivalency.icon = cFav

      carbon_equivalency.save()
      
      return carbon_equivalency, None

    except Exception as e:
      capture_message(str(e), level="error")
      return None, CustomMassenergizeError(e)`

Reply:
That's a bit more work than you need to do, and it would blow up because when you pass args into objects.create() it wants to set every field in the model, and the media isn't yet defined.  So the args.pop takes the icon out of args.  Here is the cleaned up version (which I've taken the liberty of saving in your branch:

      icon = args.pop("icon", None)      
      new_carbon_equivalency = CarbonEquivalency.objects.create(**args)

      if icon:
        cFav = Media(file=icon, name=f"{args.get('name', '')} CarbonEquivalencyFavicon")
        cFav.save()
        new_carbon_equivalency.icon = cFav
        new_carbon_equivalency.save()
      
      return new_carbon_equivalency, None